### PR TITLE
chore: pnpm migration

### PR DESCRIPTION
**Pull request by 0xmad**
# Description

- [x] Migrate docs
- [x] Migrate workflows
- [x] Migrate scripts

## Additional Notes

This PR should be merge into #987

## Related issue(s)

Related to #793 

## Confirmation

- [x] I have read and understand MACI's [contributor guidelines](https://github.com/privacy-scaling-explorations/maci/blob/dev/CONTRIBUTING.md) and [code of conduct](https://github.com/privacy-scaling-explorations/maci/blob/dev/CODE_OF_CONDUCT.md).
- [x] I have read and understand MACI's [GitHub processes](https://github.com/privacy-scaling-explorations/maci/discussions/847).
- [x] I have read and understand MACI's [testing guide](https://maci.pse.dev/docs/testing).

### CODE REVIEW

File: circuits/package.json [hunk 1]
```diff
@@ -28,9 +28,9 @@
   },
   "dependencies": {
     "circomlib": "https://github.com/weijiekoh/circomlib#ac85e82c1914d47789e2032fb11ceb2cfdd38a2b",
-    "maci-core": "^1.1.2",
-    "maci-crypto": "^1.1.2",
-    "maci-domainobjs": "^1.1.2",
+    "maci-core": "workspace:^",
```
Comment (ctrlc03) [inline circuits/package.json:31]: Have you tried publishing like this? I did once when using yarn workspaces, it published packages with this syntax and once you tried to install it from npm it failed saying it's not valid syntax

**Review by ctrlc03 — CHANGES_REQUESTED** [review]
Thank you, left one comment regarding the workspace import syntax

File: circuits/package.json (hunk 1 above)
Comment (0xmad) [reply circuits/package.json:31]: Yes, we published packages using pnpm workspaces.
https://github.com/CryptKeeperZK/crypt-keeper-extension/blob/0e4e3a0ef85f533cc35c6174d8a6c25492d1ec33/packages/providers/package.json#L39

I can try to publish it locally to check if it works.

**Review by ctrlc03 — APPROVED** [review]
Thank you!

**Comment by samajammin** [comment]
Nice @0xmad - excited to have this!

**Pull request merged**